sdk/create-dapp/templates/react-e2e-counter/src/CreateCounter.tsx

```diff
@@ -0,0 +1,58 @@
+import { TransactionBlock } from "@mysten/sui.js/transactions";
+import { Button, Container } from "@radix-ui/themes";
+import { useCallback } from "react";
+import { PACKAGE_ID } from "./constants";
+import {
+  useSignAndExecuteTransactionBlock,
+  useSuiClient,
+} from "@mysten/dapp-kit";
+
+export function CreateCounter({
+  onCreated,
+}: {
+  onCreated: (id: string) => void;
+}) {
+  const suiClient = useSuiClient();
+  const { mutateAsync: signAndExecute } = useSignAndExecuteTransactionBlock({});
+
+  const create = useCallback(() => {
```

Comment:
nit: might be nice to remove the `useCallback` because it makes the example code simpler and it doesn't seem like there's a huge benefit to memo-izing the function here